src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -18,9 +19,9 @@ public class WebSocketsTransport : IHttpTransport
         private readonly WebSocketOptions _options;
         private readonly ILogger _logger;
         private readonly Channel<byte[]> _application;
-        private readonly string _connectionId;
+        private readonly ConnectionContext _connection;
```

Comment:
The problem is that we don't know the mode when the transport is created. We only know it after receiving the first message from the client (negotiation) which contains the protocol to use.